packages/aws-cdk-lib/pipelines/lib/helpers-internal/pipeline-graph.ts

```diff
@@ -344,8 +344,9 @@ export class PipelineGraph {
       const id = stackAsset.assetType === AssetType.FILE
         ? (this.singlePublisher ? 'FileAsset' : `FileAsset${++this._fileAssetCtr}`)
         : (this.singlePublisher ? 'DockerAsset' : `DockerAsset${++this._dockerAssetCtr}`);
+      const displayName = this.singlePublisher ? id : stackAsset.displayName;
```

Comment:
I am not too fond of depending on the id for displayname, as a maintainer assumption would be that that is an implementation detail. What about:
```suggestion
      const getDisplayName: (type: AssetType): string = (type) => {
        switch(type) {
            case AssetType.FILE: return 'FileAsset(s)';
            case AssetType.DOCKER_IMAGE: return 'DockerAsset(s)';
        };
      }
      const displayName = this.singlePublisher ? getDisplayName(stackAsset.assetType) : stackAsset.displayName;
```

Reply:
Unfortunately, `FileAsset(s)` was replaced by `FileAsset_s_` :(
```
  CHANGED    pipelines/test/integ.newpipeline-single-publisher 3.847s
      Resources
[~] AWS::CodePipeline::Pipeline Pipeline9850B417
 └─ [~] Stages
     └─ @@ -121,7 +121,7 @@
        [ ]     "Name": "Synth_Output"
        [ ]   }
        [ ] ],
        [-] "Name": "FileAsset",
        [+] "Name": "FileAsset_s_",
        [ ] "RoleArn": {
        [ ]   "Fn::GetAtt": [
        [ ]     "PipelineCodeBuildActionRole226DB0CB",
```

Reply:
> Thanks for pointing out. But `switch-case` seems a bit long since `id` uses `?:`...

That might be, but it improves maintainability, since if you add a 3rd asset type, the switch code will stop compiling, while the other will happily assume that the new assets are just DockerAsset, and likely be a bug, unless someone finds all such places.

Reply:
Ok. Fixed.